register.go

```diff
@@ -2,14 +2,20 @@ package metrics
 
 import "github.com/prometheus/client_golang/prometheus"
 
+// Registrerer gets the registerer being used to register
+// and deregister namespaces, metrics, etc with prometheus.
+func Registrerer() prometheus.Registerer {
```

Comment:
Actually, I just looked into it more and it can be solved in another way. It seems most libraries like to register metrics in `init()` w/ private vars (etcd in swarmkit for example). What can be done if those need to be exposed in say debug mode is put the use `prometheus.Gathers` (array of Gathererrs) that supports combining metric families across the array.